bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/CloudBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+        var response = api.setValueForProperty(dsn, propertyName, value);
+        if (response.failed()) {
+            logger.error("Cannot set property {} on device {} to value {} using salusClient\n{}", propertyName, dsn,
+                    value, response.error());
+            devicePropertiesCache.invalidate(dsn);
+            return false;
+        }
+        var setValue = response.body();
+        if (setValue != null
+                && (!(setValue instanceof Boolean) && !(setValue instanceof String) && !(setValue instanceof Number))) {
+            logger.warn(
+                    "Cannot set value {} ({}) for property {} on device {} because it is not a Boolean, String, Long or Integer",
+                    setValue, setValue.getClass().getSimpleName(), propertyName, dsn);
+            return false;
+        }
+        var property = devicePropertiesCache.get(dsn).stream().filter(prop -> prop.getName().equals(propertyName))
+                .findFirst();
+        if (property.isEmpty()) {
+            var simpleName = setValue != null ? setValue.getClass().getSimpleName() : "<null>";
+            logger.warn(
+                    "Cannot set value {} ({}) for property {} on device {} because it is not found in the cache. Invalidating cache",
+                    setValue, simpleName, propertyName, dsn);
+            devicePropertiesCache.invalidate(dsn);
+            return false;
+        }
+        var prop = property.get();
+        if (setValue == null) {
+            prop.setValue(null);
+            return true;
+        }
+        if (setValue instanceof Boolean b && prop instanceof DeviceProperty.BooleanDeviceProperty boolProp) {
+            boolProp.setValue(b);
+            return true;
+        }
+        if (setValue instanceof String s && prop instanceof DeviceProperty.StringDeviceProperty stringProp) {
+            stringProp.setValue(s);
+            return true;
+        }
+        if (setValue instanceof Number l && prop instanceof DeviceProperty.LongDeviceProperty longProp) {
+            longProp.setValue(l.longValue());
+            return true;
+        }
+
+        logger.warn(
+                "Cannot set value {} ({}) for property {} ({}) on device {} because value class does not match property class",
+                setValue, setValue.getClass().getSimpleName(), propertyName, prop.getClass().getSimpleName(), dsn);
+        return false;
+    }
+
+    @Override
+    public SortedSet<Device> findDevices() {
+        var api = this.salusApi;
+        if (api == null) {
+            logger.error("Cannot find devices because salusClient is null");
+            return emptySortedSet();
+        }
+        logger.debug("Finding devices using salusClient");
+        var response = api.findDevices();
+        if (response.failed()) {
+            logger.error("Cannot find devices using salusClient\n{}", response.error());
```

Comment:
```suggestion
            logger.warn("Cannot find devices using salusClient\n{}", response.error());
```
as it looks like there is an api call, would this be a situation that thing Ting should go offline